src/pin_layout.rs

```diff
@@ -363,3 +376,19 @@ fn create_pin_view_side(
 
     (pin_option, pin_name, pin_arrow, pin_button)
 }
+
+fn hardware_view(gpio: &Gpio) -> Element<'static, Message> {
```

Comment:
You could maybe pass in &HardwareDescriptor (&gpio.hardware_description) to only pass in the data required?

Reply:
sure, will do that